core/replica.go

```diff
@@ -51,6 +52,33 @@ type Replica struct {
 	handleStream messageStreamHandler
 }
 
+// GetReplica creates a new instance of replica node
+func GetReplica(id uint32, configer api.Configer, stack Stack, handler api.ProtocolHandler, logger *logging.Logger) (*Replica, error) {
+	n := configer.N()
+	f := configer.F()
+
+	if n < 2*f+1 {
+		return nil, fmt.Errorf("%d nodes is not enough to tolerate %d faulty", n, f)
+	}
+
+	replica := &Replica{
+		id: id,
+		n:  n,
+
+		stack: stack,
+
+		log: messagelog.New(),
```

Comment:
This field is used in `replica.start`, but it is not accessible by `handler`.

Reply:
It was already like this in the master repository.

Reply:
Yes, but generic connectivity should also allow to somehow distribute messages generated by `ProtocolHandler`.